Surepetcare config flow & library update

benleb opened the pull request:
replaced

benleb committed:
remove deprecated config options

benleb committed:
remove deprecated config options

benleb committed:
Merge branch 'dev' of https://github.com/home-assistant/core into surepetcare-fixes-and-features

* 'dev' of https://github.com/home-assistant/core:
  [ci skip] Translation update
  Surepetcare, Use entity class vars and some clean up (#52205)
  Fix habitica regression (#52097)
  Clean up strings.json (#52202)
  Correct keyerror exception. (#52150)

cgarwood commented:
Before you get too deep in this PR, plan to split it up into a few smaller PRs if possible. Smaller PRs are much easier to review and faster to get merged 👍

benleb committed:
Merge branch 'dev' of https://github.com/home-assistant/core into surepetcare-fixes-and-features

* 'dev' of https://github.com/home-assistant/core: (318 commits)
  [ci skip] Translation update
  Replace custom listener with helper in ReCollect Waste (#52445)
  Abort existing reauth flow on entry removal (#52407)
  Fix Statistics recorder migration order (#52449)
  Fix Fritz call deflection list (#52443)
  Add static typing to devolo_home_control (#52396)
  Remove redundant property definitions in ReCollect Waste (#52368)
  Demo: remove deprecated switch entity properties (#52424)
  Add sensor platform to goalzero (#49835)
  Import track_new_devices and scan_interval from yaml for nmap_tracker (#52409)
  Enable basic type checking for Tasmota (#52435)
  Correct recorder table arguments (#52436)
  Avoid duplicated database queries when fetching statistics (#52433)
  Fix typo in forecast_solar strings (#52430)
  Upgrade aioimaplib to 0.9.0 (#52422)
  Drop statistic_id and source columns from statistics table (#52417)
  Reject trusted network access from proxies (#52388)
  Remove boilerplate code in favour of attributes in Netatmo integration (#52395)
  Add rainbird set_rain_delay service (#52369)
  Add update listener to Coinbase (#52404)
  ...

benleb committed:
Merge branch 'dev' of https://github.com/home-assistant/core into surepetcare-fixes-and-features

* 'dev' of https://github.com/home-assistant/core: (21 commits)
  Migrate AirVisual `air_quality` platform to `sensor` platform (#52349)
  Migrate GIOS air_quality platform to sensor (#52295)
  Remove empty hosts and excludes from nmap configuration (#52489)
  Use entity class attributes for airnow (#52502)
  Bump HAP-python to 3.5.1 (#52508)
  Add basic typing to ezviz camera platform (#52492)
  Update devolo-home-control-api (#52497)
  Mark entities for dead zwave_js nodes as unavailable (#48017)
  [ci skip] Translation update
  Deprecate YAML config for Ambient PWs (2021.9.0 removal) (#52459)
  Enable basic type checking for script (#52476)
  Remove redundant property definitions in Tile (#52448)
  Remove redundant property definitions in RainMachine (#52456)
  Remove redundant property definitions in SimpliSafe (#52458)
  Replace custom listener with helper in SimpliSafe (#52457)
  Enable basic type checking for trace (#52468)
  Enable basic type checking for gogogate2 (#52467)
  Enable basic type checking for climate (#52470)
  Bump aiohomekit to 0.4.1 (#52472)
  Improve typing in Sony Bravia TV integration (#52438)
  ...

benleb committed:
Merge branch 'dev' of https://github.com/home-assistant/core into surepetcare-fixes-and-features

* 'dev' of https://github.com/home-assistant/core: (110 commits)
  Bump up ZHA depdencies (#52818)
  [ci skip] Translation update
  Fix Neato parameter for token refresh (#52785)
  Add transition support to zwave_js lights (#52160)
  Add device info to Freedompro (#52715)
  Bump dependency to properly handle current and voltage not being reported on some zhapower endpoints (#52764)
  Support certain homekit devices that emit invalid JSON (#52759)
  Fix ESPHome Camera not merging image packets (#52783)
  Upgrade pymazda to 0.2.0 (#52775)
  [ci skip] Translation update
  Fix homebridge devices becoming unavailable frequently (#52753)
  Remove scale calculation for climacell cloud cover (#52752)
  Bump pylutron to 0.2.8 fixing python 3.9 incompatibility (#52702)
  Use entity class attributes for aurora (#52690)
  Fix KNX Fan features (#52732)
  Upgrade Fronius dependency to 0.5.3 (#52737)
  Esphome fix camera image (#52738)
  Add check for _client existence in modbus (#52719)
  Ignore unused keys from Sonos device properties callback (#52660)
  Use entity class attributes for apcupsd (#52662)
  ...

benleb committed:
Merge branch 'dev' of https://github.com/home-assistant/core into surepetcare-fixes-and-features

* 'dev' of https://github.com/home-assistant/core: (38 commits)
  Set device_class on temperature sensors R-Z (#52920)
  Use entity class attributes for apple_tv (#52664)
  Improve typing of Tasmota (2/3) (#52747)
  Add zeroconf discovery to Sonos (#52655)
  Handle dhcp packets without a hostname (#52882)
  Add some type hints for webhook component (#52895)
  Ignore Sonos Boost devices during discovery (#52845)
  Improve testing of option flow in Coinbase (#52870)
  Add the Trane brand to nexia (#52805)
  Bump aiohomekit to 0.5.1 to solve performance regression (#52878)
  Use entity class attributes for Bitcoin (#52887)
  Surepetcare, fix set_lock_state (#52912)
  Bump dessant/lock-threads from 2.0.3 to 2.1.1 (#52899)
  Use entity class attributes for Bizkaibus (#52888)
  Import Protocol from typing (#52848)
  Use entity class attributes for Bh1750 (#52886)
  Use entity class attributes for Blinksticklight (#52892)
  Fix Soundbar exclusion from SamsungTV (#51023)
  Prefer using xy over hs when supported by light (#52883)
  Add more data_types to modbus (#52423)
  ...

benleb committed:
add config flow and migrate to surepy>=0.7.0

benleb committed:
add config flow and migrate to surepy>=0.7.0

benleb commented:
> Before you get too deep in this PR, plan to split it up into a few smaller PRs if possible. Smaller PRs are much easier to review and faster to get merged 👍

Right... But as this is a "backport", there is a working integration I have to break up into still working PRs... 

I will migrate to the config flow configuration and change just the parts needed to get it working with the new surepy version :)

emontnemery committed:
Improve typing of Tasmota (3/3) (#52748)

teharris1 committed:
Bump pyinsteon to 1.0.11 (#52927)

frenck committed:
Upgrade sentry-sdk to 1.3.0 (#52926)

janiversen committed:
Correct Wrong "raise" in modbus validators. (#52924)

emontnemery committed:
Improve typing of Tasmota (1/3) (#52746)

cyberjunky committed:
Bump python-fireservicerota to 0.0.42 (#52807)

janiversen committed:
Add array write to turn_on/off in modbus switch/fan/light (#52582)

emontnemery committed:
Set device_class on temperature sensors F-K (#52918)

* Set device_class on temperature sensors F-K

* Fix juicenet sensor

milanmeu committed:
Use properties instead of raw data in the rituals integration (#52587)

emontnemery committed:
Set device_class on temperature sensors L-Q (#52919)

bwduncan committed:
Poll Nissan servers for battery updates (#44826)

bdraco committed:
Fix nexia thermostats humidify without dehumidify support (#52758)

bdraco committed:
Fix recorder purge with sqlite3 < 3.32.0 (#52929)

emontnemery committed:
Set device_class on temperature sensors A-E (#49524)

Co-authored-by: Franck Nijhof <git@frenck.dev>

tkdrob committed:
Use entity class attributes for bbox (#52838)

* Use entity class attributes for bbox

* tweak

tkdrob committed:
Use entity class attributes for Blockchain (#52894)

* Use entity class attributes for blockchain

* rework

* tweak

jjlawren committed:
Bump pysonos to 0.0.52 (#52934)

tkdrob committed:
Use entity class attributes for Blebox (#52890)

* Use entity class attributes for blebox

* rework

* Apply suggestions from code review

Co-authored-by: Franck Nijhof <git@frenck.dev>

Co-authored-by: Franck Nijhof <git@frenck.dev>

OttoWinter committed:
ESPHome enable static type checking (#52348)

jjlawren committed:
Use entity class attributes for Plex (#52617)

scop committed:
String formatting cleanups (#52937)

tkdrob committed:
Use entity class attributes for aqualogic (#52668)

cgtobi committed:
Add schedule selector for Netatmo (#52909)

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

Commit:
[ci skip] Translation update

raman325 committed:
Fix siren turn on parameter filtering (#52947)

* Fix siren turn on parameter filtering

* fix test

raman325 committed:
Handle FailedCommand exceptions in zwave_js WS API (#52461)

* Handle zwave-js errors in WS API

* Unsubscribe callbacks when zwave-js error is caught

* fix tests

* simplify unsub logic

* add tests

* add kwargs to be safe

* use existing msg format

* switch to generic failed command handling

* remove unneeded constant

* Update homeassistant/components/zwave_js/api.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/zwave_js/api.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* fix

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

raman325 committed:
Change behavior of Z-Wave JS services (#52941)

* Change behavior of Z-Wave JS services

* pop parameters in for loop

* Update logger message

emontnemery committed:
Minor test coverage improvement of mfi and zwave sensors (#52935)

stefano055415 committed:
Add switch support to Freedompro (#52727)

* Update Freedompro

* add test state updates from the API

* fix test switch

* fix test

bdraco committed:
Expose async_get_source_ip in the network integration (#52901)

* Expose async_get_source_ip in the network integration

* Handle source ip on disabled interface

* add coverage

raman325 committed:
Switch to using entry.async_on_remove (#52952)

emontnemery committed:
Return empty when listing statistic_ids for unsupported statistic (#52954)

Kirchoff committed:
Fix modbus climate precision configuration variable (#52651)

* Updated precision to follow the tenths, halves, whole notation used by
other home assistant climate modules. Added the precision @property so
that home assistant can handle this rounding in the frontend, rather
than in the _async_read_register() method.

* Fixed a pylinter error for periods in user-facing log messages, and
updated `precision` defaults in components/modbus/__init__.py to be
consistent with an error case, using `PRECISION_WHOLE`.

* revert changes to `precision:` configuration variable
instead, the climate `precision()` function will infer whether or not
to display in whole or tenths. halves will be unsupported, which should
be fine.

* re-added missing line that was removed

* revert change to use self._input_type instead of CALL_TYPE_REGISTER_HOLDING

raman325 committed:
Validate tone is valid when processing siren.turn_on service call (#52953)

* Validate tone is valid when processing siren.turn_on service call

* Better message

emontnemery committed:
Improve docstring for async_get_device_class_lookup (#52921)

Shulyaka committed:
Add generic hygrostat integration (#36759)

* generic_hygrostat: new integration
Co-authored-by: J. Nick Koston <nick@koston.org>
Co-authored-by: jan Iversen <jancasacondor@gmail.com>

stefano055415 committed:
Add sensor support to Freedompro (#52726)

* Update Freedompro

* Update tests/components/freedompro/test_sensor.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* add new test for sensor and add unit_of measurement and state_class

* add test state updates from the API

* optimizer code test

* optimizer code sensor

* Update homeassistant/components/freedompro/sensor.py

* Fix imports

* Update homeassistant/components/freedompro/sensor.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Update homeassistant/components/freedompro/sensor.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Update homeassistant/components/freedompro/sensor.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Co-authored-by: Erik Montnemery <erik@montnemery.com>
Co-authored-by: Franck Nijhof <frenck@frenck.nl>

emontnemery committed:
Set device_class on additional temperature sensors (#52960)

* Set device_class on additional temperature sensors

* Apply suggestions from code review

Co-authored-by: Tobias Sauerwein <cgtobi@users.noreply.github.com>

* Set device class for greeneye_monitor sensor

* Set device class for bme280 and bme680 sensor

Co-authored-by: Tobias Sauerwein <cgtobi@users.noreply.github.com>

tkdrob committed:
Use entity class attributes for automation (#52694)

* Use entity class attributes for automation

* tweak

stefano055415 committed:
Add binary_sensor support to Freedompro (#52717)

* Update Freedompro

* change _attr_unique_id with unique_id

puddly committed:
Update ZHA to support zigpy 0.34.0 device initialization (#52610)

* Handle `None` node descriptors

* Skip loading uninitialized devices

* Fix unit test incorrectly handling unset cluster `ep_attribute`

* Revert filtering devices by status during startup

tkdrob committed:
Fix flume converagerc (#52975)

rklomp committed:
Bump pysma to 0.6.4 (#52973)

emontnemery committed:
Set device class for climacell temperature sensors (#52965)

raman325 committed:
Additional fixes for siren platform (#52971)

tkdrob committed:
Use entity class attributes for aquostv (#52670)

* Use entity class attributes for aquostv

* fix

* fix

* Tweak

* tweak

tkdrob committed:
Use entity class attributes for Blackbird (#52889)

* Use entity class attributes for blackbird

* rework

* undo media_title

raman325 committed:
Add missing device classes for climacell sensors (#52979)

cyberjunky committed:
Bump python-fireservicerota to 0.0.43 (#52966)

jjlawren committed:
More graceful exception handling in Plex library sensors (#52969)

teharris1 committed:
Fix issue connecting to Insteon Hub v2 (#52970)

emontnemery committed:
Enable basic type checking for recorder (#52440)

* Enable basic type checking for recorder

* Tweak

janiversen committed:
only allow one active call in each platform. (#52823)

tkdrob committed:
Use entity class attributes for august (#52744)

peternijssen committed:
Update pyrainbird to 0.4.3 (#52990)

Commit:
[ci skip] Translation update

raman325 committed:
Make zwave_js value updated event logic more performant (#52997)

doug-hoffman committed:
Bump pyatv to 0.8.1 (#52849)

* Bump pyatv to 0.8.1

* Update apple_tv tests for new create_session location

* Update test_user_adds_unusable_device to try device with no services

pyatv >=0.8.0 considers AirPlay a valid service and no longer fails under the previous conditions.

frenck committed:
Remove flume for allowed ignore coverage violations (#53001)

frenck committed:
Remove defunct Weather Underground integration (#52999)

janiversen committed:
copy() --> deepcopy(). (#52794)

bdraco committed:
Update homekit to use network integration (#52946)

emontnemery committed:
Include future statistics in history/list_statistic_ids (#52942)

* Include future statistics in history/list_statistic_ids

* Improve tests

frenck committed:
Deprecate Lyft integration (#53005)

emontnemery committed:
Update statistics meta data on entity_id change (#52755)

raman325 committed:
Add support for Z-Wave JS siren (#52948)

* Add support for Z-Wave JS siren

* Add additional device class to discovery

* fix docstring

* Remove device class specific part of discovery schema

* rename test

* switch to entry.async_on_remove

* Fix logic based on #52971

* Use constants to unblock PR

* Add support to set volume level

* Update homeassistant/components/zwave_js/siren.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Danielhiversen committed:
Co2signal, set SCAN_INTERVAL (#53023)

* limit co2signal, wip

Signed-off-by: Daniel Hjelseth Høyer <github@dahoiv.net>

* limit co2signal

Signed-off-by: Daniel Hjelseth Høyer <github@dahoiv.net>

* limit co2signal

Signed-off-by: Daniel Hjelseth Høyer <github@dahoiv.net>

alengwenus committed:
Bump pypck to 0.7.10 (#53013)

eavanvalkenburg committed:
fix for timestamp not present in SIA (#53015)

raman325 committed:
Add missing test coverage for sirens (#53014)

raman325 committed:
Add zwave_js device triggers (#51968)

* Initial support for zwave_js device triggers

* lint

* Add node status changed trigger

* comments

* create helper function and simplify trigger logic

* simplify code

* fix exception

* remove unused type ignore

* switch to append to make future changes easier

* make exception consistent

* Add state config schema validation

* comment

* remove 0 from falsy check

* increase test coverage

* typos

* Add central scene and scene activation value notification triggers

* reorder things for readability and enumerate node statuses

* Add support for Basic CC value notifications

* fix schemas since additional fields on triggers aren't very flexible

* pylint

* remove extra logger statement

* fix comment

* dont use get when we know key will be available in dict

* tweak text

* use better schema for required extra fields that are ints

* rename trigger types to make them easier to parse

* fix strings

* missed renaming of one trigger type

* typo

* Fix strings

* reduce complexity

* Use Al's suggestion for strings

* add additional failure test cases

* remove errant logging statement

* make CC required

* raise vol.Invalid when value ID isn't legit to prepare for next PR

* Use helper function

* fix tests

* black

Commit:
[ci skip] Translation update

Danielhiversen committed:
Pylint 2.9.3 (#52972)

Co-authored-by: Marc Mueller <30130371+cdce8p@users.noreply.github.com>

Danielhiversen committed:
Use entity class attributes for Co2signal (#53032)

TomBrien committed:
Support user-defined base currency for Coinbase exchange rate sensors (#52879)

frenck committed:
Remove deprecated YAML configuration from Buienradar (#52939)

dependabot[bot] committed:
Bump actions/stale from 3.0.19 to 4 (#53042)

Bumps [actions/stale](https://github.com/actions/stale) from 3.0.19 to 4.
- [Release notes](https://github.com/actions/stale/releases)
- [Changelog](https://github.com/actions/stale/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/stale/compare/v3.0.19...v4)

---
updated-dependencies:
- dependency-name: actions/stale
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tkdrob committed:
Clean up freedompro (#52992)

eavanvalkenburg committed:
Another SIA fix for timestamp not present. (#53045)

stefano055415 committed:
Add lock support Freedompro (#52725)

* change _attr_unique_id to unique_id and resolve conflict

* add test state updates from the API

* optimizer code test

* fix test

* fix comments and add test device registry

tkdrob committed:
Use entity class attributes for bme680 (#53037)

* Use entity class attributes for bme680

* fix

tkdrob committed:
Use entity class attributes for arcam_fmj (#52675)

* Use entity class attributes for arcam_fmj

* fix

jbouwh committed:
Add MQTT humidifier platform integration (#52828)

* New mqtt humidifier platform

* Add humidifier platform

* Leave out humidity step

* Use humidity in constant for payload reset

* change TARGET_HUMIDITY_RESET payload name

* _attr_max_humidity not assigned correctly

* _target_humidity_range has a zero base

* align CONF_TARGET_HUMIDITY_MIN and MAX  with model

* shorter topics for humidity_range

* Converts float to int from template

* new humidifier abbreviations

* Add common module to support tests

* Add tests

* Addtional testing

* Always require target_humidity_command_topic

* Typo

* use available_modes to align entity model

* use avail_modes not modes to avoid conflict

* typo target_humidity_value_template

* Allign modes and templates with climate platform

* mode_state_template

* target_humidity_state_template

* Typo in platform name

* Remove humidity_range feature and common lib

* Update homeassistant/components/mqtt/humidifier.py

Use vol.In, not regex

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* black

* Update homeassistant/components/mqtt/humidifier.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Use round to convert float to target humidity

Co-authored-by: Erik Montnemery <erik@montnemery.com>

tkdrob committed:
Use entity class attributes for Bmp280 (#53036)

ludeeus committed:
Handle missing peername (#53052)

* Handle missing peername

* Add test

da-anda committed:
Fix knx expose feature not correctly falling back to default value (#53046)

bdraco committed:
Add OUIs for legacy samsungtv (#52928)

tkdrob committed:
Use entity class attributes for agent_dvr (#52501)

* Use entity class attributes for agent_dvr

* Apply suggestions from code review

Co-authored-by: Milan Meulemans <milan.meulemans@live.be>

* rework

Co-authored-by: Milan Meulemans <milan.meulemans@live.be>

tkdrob committed:
Use entity class attributes for Bloomsky (#53030)

Co-authored-by: Tobias Sauerwein <cgtobi@users.noreply.github.com>

tkdrob committed:
Use entity class attributes for androidtv (#52531)

* Use entity class attributes for androidtv

* fix

* fix pylint

* fix

balloob committed:
Expose Spotify as a service (#53063)

balloob committed:
Generate const files for config flow scaffolds (#53064)

vlebourl committed:
Increase polling interval to prevent reaching daily limit (#53066)

* increase polling interval to prevent reaching daily limit

* update test accordingly

GitHub-Action committed:
[ci skip] Translation update

tkdrob committed:
Fix google test coverage (#53060)

rklomp committed:
Use Utility Meter configured name as friendly name (#53051)

emontnemery committed:
Add light white parameter to light/services.yaml (#53075)

Shutgun committed:
Rewrite mocking in devolo Home Control tests (#53011)

* Rework mocking

* Instantiate properties

hmmbob committed:
Update name from "generic" to "generic camera" (#53080)

stefano055415 committed:
Add cover support to Freedompro (#52723)

* Update Freedompro

* add new test and fix code

* fix test and add support open and close

tkdrob committed:
Revert "Fix google test coverage (#53060)" (#53085)

This reverts commit 4075e8373264a4fdc6b0a6949ce5f27d953ca363.

AaronDavidSchneider committed:
Replace fritz profile switches by per device parental control switches (#52721)

* removes old profile switches and add new switches based on new method

* use Ellipsis instead of pass

* refactor async_add_profile_switches

* - add forgotten update_ha_state
- add notimplemtederror for devicebase

* Update homeassistant/components/fritz/common.py

Co-authored-by: J. Nick Koston <nick@koston.org>

* Update homeassistant/components/fritz/common.py

Co-authored-by: J. Nick Koston <nick@koston.org>

* comments

* fix for devices that were not connected

* Update homeassistant/components/fritz/common.py

Co-authored-by: J. Nick Koston <nick@koston.org>

* Update homeassistant/components/fritz/switch.py

Co-authored-by: J. Nick Koston <nick@koston.org>

Co-authored-by: J. Nick Koston <nick@koston.org>

stefano055415 committed:
Add fan support to Freedompro (#52724)

* Update Freedompro

* Update Freedompro fix async_turn_on

* fix test end fix comments

* add property is_on

* add percent to fan freedompro

* fix name rotationSpeed to rotation_speed

* fix code SUPPORT_SET_SPEED

p4p3r committed:
Add On/Off as target values for zwave_js cover stop action (#52881)

* Add On/Off as target values for stop cover

Certain ZWave Cover devices use On/Off instead of the more common
Open/Close and Up/Down targets for movement.
Adding On/Off to the targets used to stop the cover during movement.
Fixes issue #51963

* Add test for updated zwave_js stop cover logic

bouwew committed:
Update Plugwise Config Flow (#47563)

Co-authored-by: Tom Scholten <git@scholten.nu>

chemelli74 committed:
Fix units for Fritz network sensors (#53026)

tkdrob committed:
Use entity class attributes for Bond (#53055)

jjlawren committed:
Log source of discovery in Sonos (#53101)

GitHub-Action committed:
[ci skip] Translation update

kpine committed:
Replace local Barrier CC constants with library enums (#53109)

Bre77 committed:
Bump library version for Advantage Air (#52813)

* Bump library version

* Bump version to 0.2.5

* Add tests to cover this edge case

benleb committed:
Merge branch 'surepetcare-fixes-and-features' of github.com:benleb/core into surepetcare-fixes-and-features

* 'surepetcare-fixes-and-features' of github.com:benleb/core:
  add config flow and migrate to surepy>=0.7.0
  remove deprecated config options

homeassistant commented:
Hi @GitHub-Action, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#52206).

Once you do that we will be able to review and accept this pull request.

Thanks!
